.github/workflows/rust.yml

```
@@ -0,0 +1,122 @@
+# SPDX-FileCopyrightText: Yalan Zhang <yalzhang@redhat.com>
+# SPDX-FileCopyrightText: Timothée Ravier <tim@siosm.fr>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+# Inspired by https://github.com/coreos/repo-templates
+
+name: "Rust"
+on:
+  pull_request:
+    branches:
+      - "main"
+permissions:
+  contents: "read"
+
+# Don't waste job slots on superseded code
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+  # Pinned toolchain for linting
+  ACTIONS_LINTS_TOOLCHAIN: 1.85.0
```

this would need to be
```suggestion
  ACTIONS_LINTS_TOOLCHAIN: 1.88.0
```
**or** we relax the MSRV back down to 1.85, #14 removed the constructs that required 1.88

---

Let's start with 1.85.0 as that's what we use in CoreOS as that's in some version of RHEL. We can bump it back later.